src/lib/mathlib/math/TrajMath.hpp

```diff
@@ -45,23 +45,25 @@ namespace math
 namespace trajectory
 {
 
-/* Compute the maximum possible speed on the track given the remaining distance,
- * the maximum acceleration and the maximum jerk.
+/* Compute the maximum possible speed on the track given the desired speed,
+ * remaining distance, the maximum acceleration and the maximum jerk.
  * We assume a constant acceleration profile with a delay of 2*accel/jerk
  * (time to reach the desired acceleration from opposite max acceleration)
- * Equation to solve: 0 = vel^2 - 2*accel*(x - vel*2*accel/jerk)
+ * Equation to solve: vel_final^2 = vel_initial^2 - 2*accel*(x - vel_initial*2*accel/jerk)
  *
  * @param jerk maximum jerk
  * @param accel maximum acceleration
  * @param braking_distance distance to the desired stopping point
  *
  * @return maximum speed
  */
-inline float computeMaxSpeedFromBrakingDistance(const float jerk, const float accel, const float braking_distance)
+inline float computeMaxSpeedFromDistance(const float jerk, const float accel, const float braking_distance,
+		const float final_velocity)
 {
-	float b =  4.0f * accel * accel / jerk;
-	float c = - 2.0f * accel * braking_distance;
-	float max_speed = 0.5f * (-b + sqrtf(b * b - 4.0f * c));
+	auto sqr = [](float f) {return f * f;};
```

Comment:
Is it according to convention to use that lambda auto stuff? I'm fine with it as long as we don't overdo it.

Reply:
For a quick lamba, auto is the standard. If it starts getting more involved it is better to start specifying the return type etc. but the type names for lambdas are so long if you have a few parameters that it gets hard to read.

IMO this is really readable, but it could also be a freestanding function up at the top in an anonymous namespace if you think that's better, or I could go back to just multiplying it with itself inline like it was before.

Reply:
No it's fine, you can leave it.